### PR TITLE
Multithreaded character/music loading

### PR DESCRIPTION
**Pull request by in1tiate**
quick and dirty, about a 33% decrease in loading times here at the cost of slightly less informative loading screens

### CODE REVIEW

File: include/aoapplication.h [hunk 1]
```diff
@@ -78,6 +83,16 @@ class AOApplication : public QApplication {
   void ms_packet_received(AOPacket *p_packet);
   void server_packet_received(AOPacket *p_packet);
 
+  QMutex char_mutex;
+  QMutex music_mutex;
+  QWaitCondition charsLoaded;
+  QWaitCondition musicLoaded;
```
Comment (oldmud0) [inline include/aoapplication.h:89]: you don't need these monitors

File: src/packet_distribution.cpp [hunk 2]
```diff
@@ -455,11 +389,24 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
     if (lobby_constructed)
       w_courtroom->append_ms_chatmessage("", w_lobby->get_chatlog());
 
+    char_mutex.lock();
+    if (char_loader.isRunning())
+        charsLoaded.wait(&char_mutex);
+    char_mutex.unlock();
```
Comment (oldmud0) [inline src/packet_distribution.cpp:395]: use `charsLoaded.waitForFinished()`, but note that both your code and waitForFinished are going to block the main thread which will look like freezing.

so, maybe what you really want is a `QFutureWatcher` where when both futures are done, you then mark the loading process as finished.

however, at second glance, I'm not really seeing any I/O intensive parts in either of those two "loading" functions?

File: src/packet_distribution.cpp [hunk 3]
```diff
@@ -338,35 +341,7 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
...
-        f_char.description = sub_elements.at(1);
-
-      // temporary. the CharsCheck packet sets this properly
-      f_char.taken = false;
-
-      ++loaded_chars;
-
-      w_lobby->set_loading_text(tr("Loading chars:\n%1/%2")
-                                    .arg(QString::number(loaded_chars))
-                                    .arg(QString::number(char_list_size)));
```
Comment (oldmud0) [inline src/packet_distribution.cpp:358]: I think this was your bottleneck, not the loop itself

**Review by oldmud0 — CHANGES_REQUESTED** [review]
going through some old open PRs. don't think this needs multithreading actually

**Comment by in1tiate** [comment]
dont think this is really necessary after all, closing

**Pull request closed without merging**